### DISCUSSION TITLE
Return a query result without or with a null `data` field

### DISCUSSION THREAD

**Discussion by zizhengtai** [ExpediaGroup/graphql-kotlin#1262] [Q&A]
Hi, how can I return a query result that looks like either of the following from a Kotlin Spring WebFlux server?

1. Errors without data:
```
{
  "errors": [...]
}
```

2. Errors with null data:
```
{
  "data": null,
  "errors": [...]
}
```

Currently, if I do something like this:
```
@Component
class FooQuery : Query {
  fun foo(): Foo {
    throw ...
  }
}
```
I'll get an error complaining that I returned a null result while the return type of the query is specified as `Foo!`. But according to the GraphQL specs, the nullness should just propagate up and make `data` null.

**Comment by dariuszkuc** [comment]
Hello 👋 
Take a look at https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/execution/exceptions. You might also want to update your schema type to be nullable (i.e. `fun foo(): Foo?`).

**Reply by zizhengtai** [reply]
Hi @dariuszkuc , I have tried both of these approaches. Using `fun foo(): Foo?` causes the result to look like this when an exception is thrown, even if I don't explicitly set the data field:
```
{
  "data": {    // I want just `"data": null` instead
    "foo": null
  },
  "errors": [...]
}
```

I've also tried providing my own exception handler, which results in the same behavior anyway. I.e., I'm not able to make the `"data"` field itself `null` or completely gone.

**Reply by smyrick** [reply]
If you throw an exception in your schema code without handling any errors, the `data` field will be missing

**Reply by zizhengtai** [reply]
> If you throw an exception in your schema code without handling any errors, the `data` field will be missing

In this case I get another error in the `"errors"` list that complains about `"foo"` being null when it's schema is nonnullable. If I make the schema nullable (i.e. `fun foo(): Foo?`), then `"data"` gets populated again.

**Reply by dariuszkuc** [reply]
Indeed, I missed the part of `"data": null` vs `"data" { "foo": null}`. Currently it is not possible to have explicit null value for `data`. This is an optimization from Jackson (https://github.com/ExpediaGroup/graphql-kotlin/blob/master/servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/types/GraphQLServerRequest.kt#L39).

Do you mind elaborating on your use case and why do you need it?
